### PR TITLE
dellos module doc updates for network_cli

### PR DESCRIPTION
**Pull request by ryanmerolle**
##### SUMMARY
Added the Dell OS Modules to the Advanced Topics with Ansible for Network Automation documentation

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
docsite

##### ANSIBLE VERSION
2.7

### CODE REVIEW

**Comment by ryanmerolle** [comment]
cc @gundalow , @dharmabumstead , @ganeshrn, @trishnaguha, @skg-net, & @abirami-n

Sorry for the noise.

**Comment by skg-net** [comment]
@ryanmerolle Thanks for PR. FYI, OS10 support NETCONF connection plugin as well. Either you can fix it or we will fix it after this PR.

**Comment by ryanmerolle** [comment]
@skg-net I'll let you document http api for OS10 given this was not documented in your latest release notes, and I do not have  a OS10 device to test enabling httpapi via a new switch.

File: docs/docsite/rst/network/user_guide/platform_dellos10.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
+.. _dellos10_platform_options:
+
+***************************************
+Dell OS10 Platform Options
+***************************************
+
+OS10 supports Enable Mode (Privilege Escalation). This page offers details on how to use Enable Mode on OS10 in Ansible 2.6.
```
Comment (samccann) [inline docs/docsite/rst/network/user_guide/platform_dellos10.rst]:
```suggestion
OS10 supports Enable Mode (Privilege Escalation). This page offers details on how to use Enable Mode on OS10 in Ansible.
```

File: docs/docsite/rst/network/user_guide/platform_dellos10.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+| | **Enable Mode**         | | supported - use ``ansible_become: yes``     |
+| | (Privilege Escalation)  | | with ``ansible_become_method: enable``      |
+| |                         | | and ``ansible_become_pass:``                |
++---------------------------+-----------------------------------------------+
+| **Returned Data Format**  | ``stdout[0].``                                |
++---------------------------+-----------------------------------------------+
+
+For legacy playbooks, OS10 still supports ``ansible_connection: local``. We recommend modernizing to use ``ansible_connection: network_cli`` as soon as possible.
+
+Using CLI in Ansible 2.6
```
Comment (samccann) [inline docs/docsite/rst/network/user_guide/platform_dellos10.rst]:
```suggestion
Using CLI in Ansible
```

File: docs/docsite/rst/network/user_guide/platform_dellos10.rst [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+Example CLI Task
+----------------
+
+.. code-block:: yaml
+
+   - name: Backup current switch config (dellos10)
+     dellos10_config:
+       backup: yes
+     register: backup_dellso6_location
```
Comment (samccann) [inline docs/docsite/rst/network/user_guide/platform_dellos10.rst]:
```suggestion
     register: backup_dellos10_location
```

File: docs/docsite/rst/network/user_guide/platform_dellos6.rst [hunk 4]
```diff
@@ -0,0 +1,70 @@
+.. _dellos6_platform_options:
+
+***************************************
+Dell OS6 Platform Options
+***************************************
+
+OS6 supports Enable Mode (Privilege Escalation). This page offers details on how to use Enable Mode on OS6 in Ansible 2.6.
```
Comment (samccann) [inline docs/docsite/rst/network/user_guide/platform_dellos6.rst]:
```suggestion
OS6 supports Enable Mode (Privilege Escalation). This page offers details on how to use Enable Mode on OS6 in Ansible.
```

File: docs/docsite/rst/network/user_guide/platform_dellos6.rst [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+| | **Enable Mode**         | | supported - use ``ansible_become: yes``     |
+| | (Privilege Escalation)  | | with ``ansible_become_method: enable``      |
+| |                         | | and ``ansible_become_pass:``                |
++---------------------------+-----------------------------------------------+
+| **Returned Data Format**  | ``stdout[0].``                                |
++---------------------------+-----------------------------------------------+
+
+For legacy playbooks, OS6 still supports ``ansible_connection: local``. We recommend modernizing to use ``ansible_connection: network_cli`` as soon as possible.
+
+Using CLI in Ansible 2.6
```
Comment (samccann) [inline docs/docsite/rst/network/user_guide/platform_dellos6.rst]:
```suggestion
Using CLI in Ansible
```

File: docs/docsite/rst/network/user_guide/platform_dellos9.rst [hunk 6]
```diff
@@ -0,0 +1,70 @@
+.. _dellos9_platform_options:
+
+***************************************
+Dell OS9 Platform Options
+***************************************
+
+OS9 supports Enable Mode (Privilege Escalation). This page offers details on how to use Enable Mode on OS9 in Ansible 2.6.
```
Comment (samccann) [inline docs/docsite/rst/network/user_guide/platform_dellos9.rst]:
```suggestion
OS9 supports Enable Mode (Privilege Escalation). This page offers details on how to use Enable Mode on OS9 in Ansible.
```

File: docs/docsite/rst/network/user_guide/platform_dellos9.rst [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+| | **Enable Mode**         | | supported - use ``ansible_become: yes``     |
+| | (Privilege Escalation)  | | with ``ansible_become_method: enable``      |
+| |                         | | and ``ansible_become_pass:``                |
++---------------------------+-----------------------------------------------+
+| **Returned Data Format**  | ``stdout[0].``                                |
++---------------------------+-----------------------------------------------+
+
+For legacy playbooks, OS9 still supports ``ansible_connection: local``. We recommend modernizing to use ``ansible_connection: network_cli`` as soon as possible.
+
+Using CLI in Ansible 2.6
```
Comment (samccann) [inline docs/docsite/rst/network/user_guide/platform_dellos9.rst]:
```suggestion
Using CLI in Ansible
```

File: docs/docsite/rst/network/user_guide/platform_dellos9.rst [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+Example CLI Task
+----------------
+
+.. code-block:: yaml
+
+   - name: Backup current switch config (dellos9)
+     dellos9_config:
+       backup: yes
+     register: backup_dellso6_location
```
Comment (samccann) [inline docs/docsite/rst/network/user_guide/platform_dellos9.rst]:
```suggestion
     register: backup_dellos9_location
```

File: docs/docsite/rst/network/user_guide/platform_index.rst [hunk 9]
```diff
@@ -44,6 +47,12 @@ Settings by Platform
 +-------------------+-------------------------+----------------------+----------------------+------------------+------------------+
 | Cisco NX-OS*      | ``nxos``                | in v. >=2.5          | N/A                  | in v. >=2.6      |  in v. >=2.4     |
 +-------------------+-------------------------+----------------------+----------------------+------------------+------------------+
+| Dell OS6*         | ``dellos6``             | in v. >=2.5          | N/A                  | N/A .            |  in v. >=2.3     |
```
Comment (samccann) [inline docs/docsite/rst/network/user_guide/platform_index.rst]:
```suggestion
| Dell OS6          | ``dellos6``             | in v. >=2.5          | N/A                  | N/A .            |  in v. >=2.3     |
```

File: docs/docsite/rst/network/user_guide/platform_index.rst [hunk 10]
```diff
@@ -44,6 +47,12 @@ Settings by Platform
...
++-------------------+-------------------------+----------------------+----------------------+------------------+------------------+
+| Dell OS9*         | ``dellos9``             | in v. >=2.5          | N/A                  | N/A .            |  in v. >=2.3     |
```
Comment (samccann) [inline docs/docsite/rst/network/user_guide/platform_index.rst]:
```suggestion
| Dell OS9          | ``dellos9``             | in v. >=2.5          | N/A                  | N/A .            |  in v. >=2.3     |
```

File: docs/docsite/rst/network/user_guide/platform_index.rst [hunk 11]
```diff
@@ -44,6 +47,12 @@ Settings by Platform
...
++-------------------+-------------------------+----------------------+----------------------+------------------+------------------+
+| Dell OS10*        | ``dellos10``            | in v. >=2.5          | N/A                  | N/A .            |  in v. >=2.3     |
```
Comment (samccann) [inline docs/docsite/rst/network/user_guide/platform_index.rst]:
```suggestion
| Dell OS10         | ``dellos10``            | in v. >=2.5          | N/A                  | N/A .            |  in v. >=2.3     |
```

File: docs/docsite/rst/network/user_guide/platform_index.rst (hunk 9 above)
Comment (samccann) [reply docs/docsite/rst/network/user_guide/platform_index.rst]: The * means maintained by the Ansible Network team :-)

**Review by samccann — CHANGES_REQUESTED** [review]
Thanks for this PR!  A few nits (we're removing the '2.6' references in these guides).

**Comment by ryanmerolle** [comment]
@samccann I think I merged all your suggested changes.  Please review.

**Review by samccann — APPROVED** [review]
LGTM

**Comment by samccann** [comment]
@ryanmerolle @skg-net docs generate fine.  Is there any other gating factor here from Dell before we merge?

**Pull request merged**